### ISSUE TITLE
Nvidia offers a lot of confusing drivers

### ISSUE THREAD

**Issue by codygarver** [elementary/appcenter#247]
Nvidia support [multiple driver releases](https://devtalk.nvidia.com/default/topic/533434/linux/current-graphics-driver-releases/post/3751724/#3751724) for different hardware, which looks overwhelming when presented by AppCenter.

It would be best to only show the relevant driver.
![appcenter-nvidia-drivers](https://cloud.githubusercontent.com/assets/2166391/25497616/d3e07e50-2b4a-11e7-9fa6-0b63d5b6b362.png)

**Comment by codygarver** [comment]
```
afaik the ubuntu-drivers-common contains the logic to propose the correct driver based on the pci-id of the installed card
```
--@ricotz

**Comment by sudopluto** [comment]
yeah when you run ubuntu-drivers autoinstall, ubuntu drivers manages to select the relevant packages and install.

**Comment by sudopluto** [comment]
Coming back to this. I think I can give this a go. Looking at the code, it just uses ubuntu-drivers list, which auto sorts the available drivers per device. I will try to learn some vala, but I propose one of two options:

1) add the first driver in ubuntu-drivers list that has a new stub (stub = intel, nvidia, amd)

2) use ubuntu-drivers devices, and use == to identify a new device, and when one is found, select the first driver in the driver list

**Comment by sudopluto** [comment]
== /sys/devices/pci0000:00/0000:00:01.1/0000:02:00.0 ==
modalias : pci:v000010DEd0000139Bsv00001028sd00000706bc03sc02i00
vendor   : NVIDIA Corporation
model    : GM107M [GeForce GTX 960M]
driver   : nvidia-384 - distro non-free
driver   : nvidia-390 - third-party free recommended
driver   : nvidia-387 - third-party free
driver   : xserver-xorg-video-nouveau - distro free builtin

== cpu-microcode.py ==
driver   : intel-microcode - distro free

**Comment by waldyrious** [comment]
Would it be feasible to also add a small note in the grey text under the driver name (or perhaps under the "Drivers" header would be more appropriate) saying why one might want to install the driver? For a non-tech-savvy user whose computer is working without issues, it might be unclear why these drivers are being recommended.

**Comment by Garbee** [comment]
Filtering it down to just one option isn't a good idea. Access to all of them is necessary. I've hit a few bugs on newer drivers which required me to rollback to older ones to get the OS stable again.

Perhaps this type of UX could be done to help the situation:

1. Reverse ordering so they flow newest to oldest.
1. Have some text displayed for the recommended version.
1. Only show the latest/recommended by default. Hide the rest under a collapse menu titled "Alternative NVIDIA Drivers" or something to that effect.
1. Inside of the collapse panel, at the top we could have a little info warning to only use older/non-recommended drivers if you're experiencing issues with the latest version.

This highlights and prioritizes the recommended/latest driver, helps the "mess" the others cause, yet still allows for users to get to them in case they experience issues.

**Comment by danrabbit** [comment]
Gonna punt this feature since we're already end of April
